### PR TITLE
Image prediction using trained model

### LINKED ISSUE #1110: How to Inference non-val images in Transfer Learning Tutorial
The transfer learning tutorial is great, however it leaves off after evaluating the accuracy of the model. My goal is to then use the model that has been created to inference other images, but have run into trouble getting the new data adhere to the correct format and shape. Do you have any insight on how I could do this?

### PR DESCRIPTION
**Pull request by HemanthSai7**
Fixes #1110 

## Description
This commit enables us to perform image prediction using the saved model. We can now select any image and obtain a prediction for its corresponding label as described in the issue.

## Checklist

- [x] The issue that is being fixed is referred to in the description (see above "Fixes #1110 ")
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included in this pull request.

cc @dreiss @jonas @pietern @svekars

### CODE REVIEW

**Comment by HemanthSai7** [comment]
Hello, the build failed on ```visualize_model_upload_image(model_conv,img_path='image_path',model_name='model_name')``` saying *no such directory image_path*. How do I fix this?

File: beginner_source/data_loading_tutorial.py [hunk 1]
```diff
@@ -266,8 +266,8 @@ def __call__(self, sample):
         h, w = image.shape[:2]
         new_h, new_w = self.output_size
 
-        top = np.random.randint(0, h - new_h + 1)
-        left = np.random.randint(0, w - new_w + 1)
+        top = np.random.randint(0, h - new_h)
+        left = np.random.randint(0, w - new_w)
```
Comment (svekars) [inline beginner_source/data_loading_tutorial.py:271]: I believe these changes should not be here

Comment (HemanthSai7) [reply beginner_source/data_loading_tutorial.py:271]: Yes, I had some conflicts and noticed this file had changes that I didn't do it. I reverted the changes and it got committed too. I believe that was some other contributor's work which I didn't notice.

File: beginner_source/data_loading_tutorial.py [hunk 2]
```diff
@@ -266,8 +266,8 @@ def __call__(self, sample):
         h, w = image.shape[:2]
         new_h, new_w = self.output_size
 
-        top = np.random.randint(0, h - new_h + 1)
-        left = np.random.randint(0, w - new_w + 1)
+        top = np.random.randint(0, h - new_h)
```
Comment (svekars) [inline beginner_source/data_loading_tutorial.py:269]:
```suggestion
        top = np.random.randint(0, h - new_h + 1)
```

File: beginner_source/data_loading_tutorial.py [hunk 3]
```diff
@@ -266,8 +266,8 @@ def __call__(self, sample):
...
+        left = np.random.randint(0, w - new_w)
```
Comment (svekars) [inline beginner_source/data_loading_tutorial.py:270]:
```suggestion
        left = np.random.randint(0, w - new_w + 1)
```

File: beginner_source/data_loading_tutorial.py [hunk 4]
```diff
@@ -292,7 +292,7 @@ def __call__(self, sample):
 
 ######################################################################
 # .. note::
-#     In the example above, `RandomCrop` uses an external library's random number generator
+#     In the example above, `RandomCrop` uses an external library's random number generator
```
Comment (svekars) [inline beginner_source/data_loading_tutorial.py:295]:
```suggestion
#     In the example above, `RandomCrop` uses an external library's random number generator
```

File: beginner_source/data_loading_tutorial.py (hunk 1 above)
Comment (svekars) [reply beginner_source/data_loading_tutorial.py]: I resolved this for you

Comment (HemanthSai7) [reply beginner_source/data_loading_tutorial.py]: Thank you! For future reference, in these situations, I have to revert to the previous commit hash right?

File: beginner_source/transfer_learning_tutorial.py [hunk 5]
```diff
@@ -339,6 +340,39 @@ def visualize_model(model, num_images=6):
...
+######################################################################
+# Save and load the model
+# ----------------------
+#
+# Here we have saved the trained model and loaded it for inference. We can 
+# now use our trained model to make predictions on our own images and analyze 
+# the results.
+#
+
+def save_and_load_model(model, model_name):
```
Comment (kit1980) [inline beginner_source/transfer_learning_tutorial.py:352]: Is this a convoluted way to do a deepcopy?
I think in this case using deepcopy is fine if it works.

**Review by kit1980 — CHANGES_REQUESTED** [review]

**Review by kit1980 — CHANGES_REQUESTED** [review]
Have you verified that this is working?
Also please address the comments and format the Python code according to PEP8.

File: beginner_source/transfer_learning_tutorial.py [hunk 6]
```diff
@@ -339,6 +340,39 @@ def visualize_model(model, num_images=6):
...
+    torch.save(model.state_dict(), model_name)
+    model.load_state_dict(torch.load(model_name))
+    return model
+
+def visualize_model_upload_image(model,model_name,img_path):
+    was_training = model.training
+    model=save_and_load_model(model,model_name)
```
[the remark this answers is not in the document]
Comment (HemanthSai7) [reply beginner_source/transfer_learning_tutorial.py:357]: Okay, I get your point. I'll make the changes as requested following PEP8 guidelines.

**Comment by HemanthSai7** [comment]
@svekars @kit1980 I have made the requested changes and also formatted the code according to PEP8.

**Review by kit1980 — APPROVED** [review]
